pkg/controllers/dashboard/hostedcluster/controller.go

```diff
@@ -108,6 +133,28 @@ func (h handler) onClusterChange(key string, cluster *v3.Cluster) (*v3.Cluster,
 	return cluster, nil
 }
 
+func (h handler) removeLegacyOperatorIfExists(provider string) error {
+	systemProject, err := project.GetSystemProject(localCluster, h.projectCache)
+	if err != nil {
+		return err
+	}
+
+	systemProjectID := ref.Ref(systemProject)
+	_, systemProjectName := ref.Parse(systemProjectID)
+
+	legacyOperatorAppName := fmt.Sprintf(legacyOperatorAppNameFormat, provider)
+	_, err = h.appCache.Get(systemProjectName, legacyOperatorAppName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// legacy app doesn't exist, no-op
+			return nil
+		}
+		return err
+	}
+
+	return h.apps.Delete(systemProjectName, legacyOperatorAppName, &metav1.DeleteOptions{})
```

Comment:
Summary from our chart: This will remove the system-chart, we need to be positive this won't delete or interrupt previous cluster objects/resources and that they get smoothly transitioned. Need to make that clear for QA to test.

Reply:
To be clear, deleting the chart does delete resources associated with it except for the CRD. It should be inconsequential since it does not delete the cluster and the resources will be installed with the new chart.